source/git-api.js

```diff
@@ -66,10 +55,10 @@ exports.registerApi = (env) => {
           return mkdirp(pathToWatch);
         }
       }).then(() => {
-        const watcher = fs.watch(pathToWatch, options || {});
-        watcher.on('change', (event, filename) => {
-          if (!filename) return;
-          const filePath = path.join(subfolderPath, filename);
+        const watcher = watch(pathToWatch, options || {});
+        watcher.on('change', (event, changedPath) => {
+          if (!changedPath) return;
+          const filePath = path.relative(socket.watcherPath, changedPath);
```

Comment:
On windows I got a file changed event for the `watcherPath` itself in which case `path.relative` returns an empty string and an exception was thrown in `ignore`.

```
Exception: TypeError
TypeError: path must not be empty
    at throwError (p:\ungit\node_modules\ignore\index.js:337:9)
    at checkPath (p:\ungit\node_modules\ignore\index.js:350:12)
    at Ignore._test (p:\ungit\node_modules\ignore\index.js:473:5)
    at Ignore.ignores (p:\ungit\node_modules\ignore\index.js:512:17)
    at p:\ungit\node_modules\ignore\index.js:516:26
    at Array.filter (<anonymous>)
    at Ignore.filter (p:\ungit\node_modules\ignore\index.js:520:29)
    at isFileWatched (p:\ungit\source\git-api.js:87:26)
    at Watcher.<anonymous> (p:\ungit\source\git-api.js:63:15)
    at Watcher.emit (events.js:223:5)
    at p:\ungit\node_modules\node-watch\lib\watch.js:305:19
    at p:\ungit\node_modules\node-watch\lib\watch.js:39:7
    at p:\ungit\node_modules\node-watch\lib\watch.js:303:7
    at p:\ungit\node_modules\node-watch\lib\watch.js:109:10
    at Array.forEach (<anonymous>)
    at Timeout.handle [as _onTimeout] (p:\ungit\node_modules\node-watch\lib\watch.js:104:24)
```

Reply:
ok, so the returned path can be `relativePath || '.'`? Or what should it be in this case?

Reply:
It should be ignored completly `if (!filePath) return;` like on the line before.